content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+```
+![dfsample5](dfsample5.png)
+```python
+df_new = df.drop(columns=["sepal_width", "sepal_length", "petal_length", "petal_width"])
+```
+```python
+df_new.head()
+```
+![newhead](df_new.png)
+
+Let's make use of `chic2_contigency` provided for us in `scipy.stats` library.
+
+It has been added to our dependencies earlier in the notebook.
+```python
+stat, p, dof, expected= chi2_contingency(df_new)
+
+print(dof)
+```
+![dof](DOF.png)
+
+We can comfortably get 3 outputs in test statistics, degree of freedom, p-value, and expected values.
+
+[Test statistic](https://online.stat.psu.edu/statprogram/reviews/statistical-concepts/chi-square-tests) is a characterized feature of significance in the Chi-square test that helps in determining how uncommon the result might be provided that null hypothesis is assumed to be true.
+
+The number of features or variables that we have access to varying is the degree of freedom. Follow the link for more information on [degree of freedom](https://byjus.com/maths/chi-square-test/).
+
+Hence, we set our probability of confidence to 95% as explained earlier in the tutorial.
+```python
+prob = 0.95
+
+alpha = 1.0 - prob
+
+print('alpha/significance level=%.3f' % alpha)
+print('P-value =%.2f' % p)
+
+if p <= alpha:
+ print('Reject the Null Hypothesis ( reject H0)')
+else: 
+ print('Accept the Null Hypothesis (fail to reject H0)')
+```
+![alphsig](alphasiglevel.png)
+
+The P-value is greater than the alpha value set at 0.05, we will fail to reject the null hypothesis(we accept the null hypothesis). This means the petal width and species of flower are not dependent, i.e. there is no relationship between them.
+
+#### T-Test
+
+The T-test is utilized when we plan to evaluate the discrepancy between the means of two groups of samples. Unlike Chi-Square Test, the T-test is used on continuous variables.
+
+It has three basic types
+
+1. **One sample t-test:** It checks whether the population has a different mean from the sample.
+The mathematical formula is:
+```
+t-test = a-BX∕N
+```
+Where;
+
+a = the observed mean group.
+
+B = the assumed population means.
+
+X = the standard deviation of the data group.
+
+N = the number of observations in the group.
+
+2. **Two-sample t-test:** It checks and compares the means of two groups that are not dependent on each other and compare with the population's means to see if there is a huge difference.
+The mathematical expression is:
+```
+t-test = Ma – (Mb*v*Na)+vNb
+```
+Ma = Data sample A's mean
+
+Mb = Data sample B's mean
+
+Na = Size of sample A
+
+Nb = Size of sample B
+
+v = Variance
+
+3. **Paired t-test:** It is used in checking and comparing the means of different samples from a group.
+Mathematically, it can be resolved with:
+```
+t-test = D - (m*xd)∕N
+```
+D = The difference between paired observations' sample mean.
+
+m = assumed mean difference.
+
+xd = standard deviation.
+
+N = the number of observations in the group.
+
+#### Performing a One-Sample T-Test
```

Comment:
Should be in H3 and the title reworded as "One-sample t-test implementation in Python". Make changes for all such occurrences.